src/components/ExperienceContainer/ExperienceContainer.jsx

```diff
@@ -1,32 +1,34 @@
 import { useState, useEffect } from "react";
 import "./ExperienceContainer.css";
+import Experience from "./Experience";
 
 export const ExperienceContainer = () => {
-  const [experience, setExperience] = useState([]);
+  const [experienceList, setExperienceList] = useState([]);
 
   useEffect(() => {
-    setExperience(mockExperience);
+    setExperienceList(mockExperience);
   }, []);
 
+  function saveExperience(idx, experience) {
+    setExperienceList((prev) => {
+      const newList = [...prev];
+      newList[idx] = experience;
+      return newList;
+    });
+  }
+
   return (
     <div className="resumeSection">
       <h2>Experience</h2>
       <div className="experienceBox">
-        {experience.map((experience) => (
-          <div className="experience" key={experience.id}>
-            <div className="content">
-              <h3>
-                {experience.company} - {experience.title}
-              </h3>
-              <p className="dates">
-                {experience.startDate} - {experience.endDate}
-              </p>
-              <p className="description">{experience.description}</p>
-            </div>
-            <div className="logo">
-              <img src={experience.logo} alt="logo" height={128} width={128} />
-            </div>
-          </div>
+        {experienceList.map((experience, idx) => (
```

Comment:
Can we have some checks for the idx bounds somewhere, I feel like this can be manipulated by adding dummy entries manually using console